core/src/main/java/com/gamification/core/repositories/UserRepository.java

```diff
@@ -5,6 +5,9 @@
 import org.springframework.stereotype.Repository;
 
 @Repository
-public interface UserRepository extends JpaRepository<User,Long> {
+public interface UserRepository extends JpaRepository<User, Long> {
 
+    User getUserById(Long id);
+
+    User getUserByUsername(String userName);
```

Comment:
username word do not need to be camel case so `String username` is going to be ok